### PR TITLE
fix(CMSIS): Fix `AFE_ADC_n_CTRL.ref_sel` enum fields and remove deprecated registers

### LINKED ISSUE #737: MAX32675: wrong bit settings in afe_adc_zero_regs.h for ref_sel
the bit settings for ref_sel case REF0P/REF0N (refer to UG7594 page 246) must be 0x2 in afe_adc_zero_regs.h (line 367)

Additionally REF2P/REF2N does not exit (lines 372ff):
#define MXC_V_AFE_ADC_ZERO_CTRL_REF_SEL_REF2P_AND_REF2N ((uint8_t)0x2UL) 
#define MXC_S_AFE_ADC_ZERO_CTRL_REF_SEL_REF2P_AND_REF2N

### PR DESCRIPTION
**Pull request by sihyung-maxim**
### Description

Fixes the `AFE_ADC_n_CTRL.ref_sel` enumerated fields.

Remove deprecated registers.

Fix #737

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

**Comment by sihyung-maxim** [comment]
This PR has been hanging for a while. This corrects the names and values of the affected register definitions.

**Comment by jdk-maxim** [comment]
ok.

I'll take a look at.

Thanks,
Jeremy
________________________________

File: Libraries/CMSIS/Device/Maxim/MAX32680/Include/afe_adc_zero_regs.h [hunk 1]
```diff
@@ -563,7 +563,7 @@ extern "C" {
  * @{
  */
 #define MXC_F_AFE_ADC_ZERO_PART_ID_REV_ID_POS          0 /**< PART_ID_REV_ID Position */
-#define MXC_F_AFE_ADC_ZERO_PART_ID_REV_ID              ((uint32_t)(0x7UL << MXC_F_AFE_ADC_ZERO_PART_ID_REV_ID_POS)) /**< PART_ID_REV_ID Mask */
+#define MXC_F_AFE_ADC_ZERO_PART_ID_REV_ID              ((uint32_t)(0x3FUL << MXC_F_AFE_ADC_ZERO_PART_ID_REV_ID_POS)) /**< PART_ID_REV_ID Mask */
```
Comment (jdk-maxim) [inline Libraries/CMSIS/Device/Maxim/MAX32680/Include/afe_adc_zero_regs.h:566]: I don't think this looks right.  Its only changed for 680 and only in this header file.  The svd is not updated to match either, so this is likely a mistake.

**Comment by jdk-maxim** [comment]
Other than the REV_ID item, this looks good to me.

Thanks!

**Review by jdk-maxim — APPROVED** [review]
Reviewed.  Slight issue with REV_ID on MAX32680.  Otherwise approved.

File: Libraries/CMSIS/Device/Maxim/MAX32680/Include/afe_adc_zero_regs.h (hunk 1 above)
Comment (sihyung-maxim) [reply Libraries/CMSIS/Device/Maxim/MAX32680/Include/afe_adc_zero_regs.h:566]: Nice catch, fixed the SVD and updated the ME16's PART_ID register. However, I think we're dropping support for the ME20, so I will leave the ME20 files as there are to match the UG.

**Pull request merged**